util/test/sub_test.py

```diff
@@ -2425,29 +2426,34 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
-                    sys.stdout.flush()
-                    p = py3_compat.Popen([utildir+'/test/computePerfStats',
-                                         perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate],
-                                         stdout=subprocess.PIPE)
-                    sys.stdout.write('%s'%(p.communicate()[0]))
-                    sys.stdout.flush()
+                    # if the program exited with a non-zero exit code, don't attempt to collect performance keys
+                    status = 0
+                    if exec_status == 0:
+                        sys.stdout.write('[Executing %s/test/computePerfStats %s %s %s %s %s %s]\n'%(utildir, perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate))
+                        sys.stdout.flush()
+                        p = py3_compat.Popen([utildir+'/test/computePerfStats',
+                                            perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate],
+                                            stdout=subprocess.PIPE)
+                        sys.stdout.write('%s'%(p.communicate()[0]))
+                        sys.stdout.flush()
 
-                    status = p.returncode
-                    if not exectimeout and not launcher_error:
-                        if status == 0:
-                            os.unlink(execlog)
-                            sys.stdout.write('%s[Success '%(futuretest))
-                        else:
-                            sys.stdout.write('%s[Error '%(futuretest))
-                        sys.stdout.write('matching performance keys for %s/%s'%
-                                        (localdir, test_filename))
-                        if status!=0:
-                            printTestVariation(compoptsnum, compoptslist,
-                                               execoptsnum, execoptslist);
-                        sys.stdout.write(']\n')
+                        status = p.returncode
+                        if not exectimeout and not launcher_error:
+                            if status == 0:
+                                os.unlink(execlog)
+                                sys.stdout.write('%s[Success '%(futuretest))
+                            else:
+                                sys.stdout.write('%s[Error '%(futuretest))
+                            sys.stdout.write('matching performance keys for %s/%s'%
+                                            (localdir, test_filename))
+                            if status!=0:
+                                printTestVariation(compoptsnum, compoptslist,
+                                                execoptsnum, execoptslist);
+                            sys.stdout.write(']\n')
+                    else:
+                        sys.stdout.write('[Error execution failed for %s]\n'%(test_name))
```

Comment:
For some reason the above error message uses `%s/%s'%(localdir, test_filename)` instead of `%s'%(test_name)`, not sure if that's for any good reason, but if `test_name` seemed to work, that's good with me.

Reply:
Hmm, I'm not sure either. I was going off of `sys.stdout.write('[Error compilation failed for %s]\n'%(test_name))` on line 1825. 

Anyway, the output looked good to me, so I'll stick with this.